### PR TITLE
Ensure that `grc` and `gcp` always have the correct ordering, regardless of how they were input

### LINKED ISSUE #253: GCP Doesn't order the commits in oldest to newest, which can lead to merge errors
After doing a very messy cherry-pick the other day, I noticed that forgit just passes in the selections in order they were selected.

This means if you have a commit history like this, that you are cherry picking from

my_target_branch 
```
commitA
commitB
commitC
```

and you used gcp to select all of them in the order B, C, A - it would be fed to `git cherry-pick` in the order B, C, A.

I think we should re-order the list in topo order, such that no matter what the order they were selected in, they would be fed in to `git cherry-pick` like:
`git cherry-pick commitC commitB commitA`

This would prevent many merge conflicts - in general folks probably want to cherry pick in order from oldest to newest.

I attempted for a while to get this to work, but to no avail. Anyone have any thoughts?

Some discoveries I've made that may be helpful:

List the commits from the $target in topo order:
```
git rev-list --topo-order --no-walk --abbrev-commit --branches $target
```

we have all of the abbreviated commits in an variable called commits
```
echo ${commits[@]}
```

We could possibly iterate over the commit in `commits`, try and match them to the rev-list and put them in a new array?

I'm missing the tools to complete this. Interested in your thoughts @wfxr and @carlfriedrich

### PR DESCRIPTION
**Pull request by cjappl**
## Check list

- [x] I have performed a self-review of my code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Description

Closes #253. Essentially the idea is this: 

- When cherry picking a group of commits, most of the time you want to cherry pick oldest to newest. This means you have the lowest chance of having a conflict or error
- when reverting a commit. You most often want to revert from newest to oldest. Similarly prevents errors

This commit does that, based on the wonderful suggestion on how to do that from @carlfriedrich in the linked ticket (#253). 

Best reviewed with turning whitespace off, as I added some in there. I also made a few small stylistic changes so these two functions mirrored each other more closely, as fundamentally they do similar things (get a list of commits, and feed them to the function).

TO TEST:

- Try to revert one commit
- Try to revert 3 commits in all orders  - the resulting order should be the same, from newest to oldest
- Try to cherry pick one commit
- Try to cherry pick 3 commits in all orders -   the resulting order should always be the same, from oldest to newest

I was also adding a line to confirm this, which may be helpful in your testing:
```
echo "${commits[@]}"
```

on lines 251 and 247.

## Type of change

- [x] Bug fix
- [x] New feature

## Test environment

- Shell
    - [ ] bash
    - [x] zsh
    - [x] fish
- OS
    - [x] Mac OS X

### CODE REVIEW

**Comment by cjappl** [comment]
Calling all the usual suspects @sandr01d @carlfriedrich @wfxr to test if you have time before this is merged. Works like a charm on my end so far and removes one of my biggest pain points!

**Comment by cjappl** [comment]
Just used this for the first time in my work day for `gcp` man is it nice!! I've been using a...*shudder*...GUI to do cherry picking because of this issue and this may save me from ever doing it again!

**Comment by cjappl** [comment]
Added comments. Will keep this open for a few days if others want to review/poke around. I'll merge it early next week if I hear no objections. Thanks for the review @carlfriedrich

**Review by sandr01d — APPROVED** [review]

**Comment by sandr01d** [comment]
> > Works great, good work! I've requested a minor change above though.
> 
> @sandr01d could you re-post your request? I don't see it anywhere

You've already addressed it in [1eac221](https://github.com/wfxr/forgit/pull/261/commits/1eac221f8c1e86bd372666dc470b0882c15b5485) :slightly_smiling_face:

**Comment by cjappl** [comment]
Ah! Nice. Good teamwork then :) I figured it out via magic.

I’ll merge this sometime soon. Thanks for the eyes.

**Pull request merged**
